src/HieDb/Create.hs

```diff
@@ -168,26 +178,34 @@ addTypeRefs db path hf ixs = mapM_ addTypesFromAst asts
         mapM_ (addTypeRef db path arr ixs (nodeSpan ast)) $ nodeType $ nodeInfo ast
       mapM_ addTypesFromAst $ nodeChildren ast
 
-{-| Adds all references from given .hie file to 'HieDb'.
+{-| Adds all references from given @.hie@ file to 'HieDb'.
 The indexing is skipped if the file was not modified since the last time it was indexed.
 -}
 addRefsFrom :: (MonadIO m, NameCacheMonad m) => HieDb -> FilePath -> m ()
 addRefsFrom c@(getConn -> conn) path = do
   time <- liftIO $ getModificationTime path
   mods <- liftIO $ query conn "SELECT * FROM mods WHERE hieFile = ? AND time >= ?" (path, time)
-  let isBoot = "boot" `isSuffixOf` path
   case mods of
     (HieModuleRow{}:_) -> return ()
-    [] -> withHieFile path $ \hf -> addRefsFromLoaded c path isBoot Nothing False time hf
+    [] -> withHieFile path $ \hf -> addRefsFromLoaded c path Nothing False time hf
 
-addRefsFromLoaded :: (MonadIO m) => HieDb -> FilePath -> Bool -> Maybe FilePath -> Bool -> UTCTime -> HieFile -> m ()
-addRefsFromLoaded db@(getConn -> conn) path isBoot srcFile isReal time hf = liftIO $ withTransaction conn $ do
+addRefsFromLoaded
+  :: MonadIO m
+  => HieDb -- ^ HieDb into which we're adding the file
+  -> FilePath -- ^ Path to @.hie@ file
+  -> Maybe FilePath -- ^ Path to .hs file from which @.hie@ file was created
+  -> Bool -- ^ Is this a real source file? I.e. does it come from user's project (as opposed to from project's dependency)?
+  -> UTCTime -- ^ The last modification time of the @.hie@ file
+  -> HieFile -- ^ Data loaded from the @.hie@ file
+  -> m ()
+addRefsFromLoaded db@(getConn -> conn) path srcFile isReal time hf = liftIO $ withTransaction conn $ do
   execute conn "DELETE FROM refs  WHERE hieFile = ?" (Only path)
   execute conn "DELETE FROM decls WHERE hieFile = ?" (Only path)
   execute conn "DELETE FROM defs  WHERE hieFile = ?" (Only path)
   execute conn "DELETE FROM typerefs WHERE hieFile = ?" (Only path)
 
-  let mod    = moduleName smod
+  let isBoot = "boot" `isSuffixOf` path
```

Comment:
It makes more sense to calculate isBoot in this function because this flag is derived from one of its input arguments (less chance for things to go wrong because from now on nobody will be able to pass inconsistent path + isBoot combination)